framework/src/timeintegrators/BDF2.C

```diff
@@ -64,6 +65,28 @@ BDF2::computeTimeDerivatives()
   }
 }
 
+void
+BDF2::computeDotProperties(Moose::MaterialDataType type, THREAD_ID tid) const
+{
+  auto material_data = _fe_problem.getMaterialData(type, tid);
+
+  auto props = material_data->getPropertiesForDot();
+  auto old_props = material_data->getOldPropertiesForDot();
+  auto older_props = material_data->getOlderPropertiesForDot();
+  auto dot_props = material_data->getDotProperties();
+
+  for (unsigned int i = 0; i < dot_props.size(); ++i)
```

Comment:
range-based `for`

Reply:
need `i` to index the other three vectors.